lib/puppet/util/windows/taskscheduler2_v1task.rb

```diff
@@ -0,0 +1,559 @@
+require 'puppet/util/windows/taskscheduler2'
+require 'puppet/util/windows/taskscheduler' # Needed for the WIN32::ScheduledTask flag constants
+
+# This class is used to manage V1 compatible tasks using the Task Scheduler V2 API
+# It is designed to be a binary compatible API to puppet/util/windows/taskscheduler.rb but
+# will only surface the features used by the Puppet scheduledtask provider
+#
+class Puppet::Util::Windows::TaskScheduler2V1Task
+  # The error class raised if any task scheduler specific calls fail.
+  class Error < Puppet::Util::Windows::Error; end
+
+  public
+  # Returns a new TaskScheduler object. If a work_item (and possibly the
+  # the trigger) are passed as arguments then a new work item is created and
+  # associated with that trigger, although you can still activate other tasks
+  # with the same handle.
+  #
+  # This is really just a bit of convenience. Passing arguments to the
+  # constructor is the same as calling TaskScheduler.new plus
+  # TaskScheduler#new_work_item.
+  #
+  def initialize(work_item = nil, trigger = nil)
+    @task = Puppet::Util::Windows::TaskScheduler2.new()
+
+    if work_item
+      if trigger
+        raise TypeError unless trigger.is_a?(Hash)
+        new_work_item(work_item, trigger)
+      end
+    end
+  end
+
+  def enum()
+    array = []
+    @task.enum_task_names(Puppet::Util::Windows::TaskScheduler2::ROOT_FOLDER,
+        include_child_folders: false,
+        include_compatibility: [Puppet::Util::Windows::TaskScheduler2::TASK_COMPATIBILITY_AT, Puppet::Util::Windows::TaskScheduler2::TASK_COMPATIBILITY_V1]).each do |item|
+      array << @task.get_task_name_from_task(item)
+    end
+
+    array
+  end
+
+  alias :tasks :enum
+
+  def validate_task_name(task_name)
+    # The Puppet provider and some other instances may pass a '.job' suffix as per the V1 API
+    # This is not needed for the V2 API so we just remove it
+    task_name = task_name.slice(0,task_name.length - 4) if task_name.end_with?('.job')
+
+    task_name
+  end
+
+  def activate(task_name)
+    raise TypeError unless task_name.is_a?(String)
+
+    full_taskname = Puppet::Util::Windows::TaskScheduler2::ROOT_FOLDER + validate_task_name(task_name)
+
+    result = @task.activate(full_taskname)
+    return nil if result.nil?
+    if @task.compatibility != Puppet::Util::Windows::TaskScheduler2::TASK_COMPATIBILITY_AT && @task.compatibility != Puppet::Util::Windows::TaskScheduler2::TASK_COMPATIBILITY_V1
+      @task.deactivate
+      result = nil
+    end
+
+    result
+  end
+
+  def delete(task_name)
+    
+    @task.delete(validate_task_name(task_name))
+  end
+
+  def run()
+    @task.run(nil)
+  end
+
+  def save(file = nil)
+    raise NotImplementedError unless file.nil?
```

Comment:
It looks like the existing provider implementation doesn't pass a `file` anyhow in `flush` at https://github.com/puppetlabs/puppet/blob/master/lib/puppet/provider/scheduled_task/win32_taskscheduler.rb#L236

This is making us wonder if `pIPersistFile.Save` is being called with a NULL string??  https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/windows/taskscheduler.rb#L291-L294